sql/core/src/main/scala/org/apache/spark/sql/expressions/udaf.scala

```diff
@@ -129,6 +129,13 @@ abstract class UserDefinedAggregateFunction extends Serializable {
         isDistinct = true)
     Column(aggregateExpression)
   }
+
+  /**
+    * The name of the UDAF. This is currently the simple name of class. This can create an
+    * [[java.lang.InternalError]] if the UDAF class was created in the REPL; override this method
+    * in these cases.
+    */
+  def name: String = getClass.getSimpleName
```

Comment:
I am not catching a ```java.lang.InternalError``` here, because I think these shouldn't be caught at all (the error could have different cause). We enable the use to redefine the name to work arround this problem.